src/components/Directory/index.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useEffect } from 'react';
+import MentorCard from './MentorCard';
+import Subheader from '../Subheader/Subheader';
+import DirectorySearchBar from './DirectorySearch';
+import useStyles from './Directory.styles';
+import EmptyDirectory from './EmptyDirectory';
+import { Typography, Tab, Tabs, useMediaQuery } from '@material-ui/core';
+import { emptySearch, directoryTabs, subheaderProperties } from './utils';
+import request from '../../utils/axiosConfig';
+
+const Directory = () => {
+  const isMobile = useMediaQuery('(max-width:1023px)');
+  const {
+    root,
+    mentorDirectory,
+    mentorListContainer,
+    directoryHeader,
+    directoryTab,
+    tabs,
+  } = useStyles({
+    isMobile,
+  });
+  const [selectedTab, setSelectedTab] = useState(0);
+  const [mentorCards, setMentorCards] = useState([]);
+  const [query, setQuery] = useState('');
+
+  const handleChange = (e, newVal) => setSelectedTab(newVal);
+
+  useEffect(() => {
+    async function fetchData() {
+      const { data } = await request.get('/api/directory');
+      setMentorCards(data);
+    }
```

Comment:
we need to handle error here; let's do a try/catch block, and set a state for error, if error then throw new Error()

Reply:
Resolved in : 947eb29

Reply:
Added a loading component when fetching data in commit # 3eb35d5

Reply:
Updated conditional statement for directory on load commit: 13938f9

Reply:
resolving this convo :+1: